code/game/objects/items/reagent_containers/glass.dm

```diff
@@ -539,6 +544,15 @@
 	to_chat(usr, SPAN_WARNING("[src] has no transfer control valve! Use a dispenser to fill it!"))
 	return
 
+/obj/item/reagent_container/glass/pressurized_canister/on_reagent_change()
+	update_icon()
+
+/obj/item/reagent_container/glass/pressurized_canister/update_icon()
+	color = COLOR_WHITE
+	if(reagents)
+		color = mix_color_from_reagents(reagents.reagent_list)
```

Comment:
Confused - do you change both the item and its contents separately?

Reply:
You mean updating the Pressurized Reagent Canister Pouch and the Pressurized Canister? Since you can smack the Pouch against a Reagent Tank to change the contents of the Canister, then take the Canister out through a chemical dispenser (or with the other PR, a right click) so you could see it having a wrong tint.